config/v1/types_ingress.go

```diff
@@ -152,8 +153,64 @@ type AWSIngressSpec struct {
 	// +kubebuilder:validation:Enum:=NLB;Classic
 	// +kubebuilder:validation:Required
 	Type AWSLBType `json:"type,omitempty"`
+
+	// networkLoadBalancerParameters holds configuration parameters for an AWS
+	// Network Load Balancer. This field is optional and permitted only if the type field is set to NLB.
+	//
+	// +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+	// +optional
+	NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
 }
 
+// AWSNetworkLoadBalancerParameters holds configuration parameters for an
+// AWS Network Load Balancer.
+type AWSNetworkLoadBalancerParameters struct {
+	// eipAllocations is a list of IDs for Elastic IP (EIP) addresses that
+	// are assigned to the Network Load Balancer.
```

Comment:
I think it makes sense to come right out and say what this field is for vs. being implicit about it.

```suggestion
	// eipAllocations is a list of IDs for Elastic IP (EIP) addresses that
	// are assigned to the  Network Load Balancer for the default IngressController.
```